lib/productions/extended-attributes.js

```diff
@@ -1,8 +1,30 @@
 import { Base } from "./base.js";
 import { ArrayBase } from "./array-base.js";
-import { list, identifiers, argument_list, autoParenter } from "./helpers.js";
+import { Token } from "./token.js";
+import { list, argument_list, autoParenter } from "./helpers.js";
 import { validationError } from "../error.js";
 
+/**
+ * @param {import("../tokeniser").Tokeniser} tokeniser
+ * @param {string} tokenName
+ */
+function tokens(tokeniser, tokenName) {
+  const toks = list(tokeniser, {
+    parser: Token.parser(tokeniser, tokenName),
+    listName: tokenName + " list"
+  });
+  if (!toks.length) {
+    tokeniser.error(`Expected ${tokenName}s but none found`);
```

Comment:
Could you move this length check to `identifiersOrStrings()? Otherwise it won't consume strings as it fails consuming identifiers first.

Reply:
Since we are getting toks after parsing them, so should I also parse it first?

Reply:
Exactly.

Reply:
Then I think token functions is of no use